File: Top 100 Codes/02-Working with Numbers/09-Decimal to Hexadecimal Conversion/DecimaltoHexdecimalConversion.py
def convert_to_hexadecimal(num):
    """
    Converts a decimal number to its hexadecimal representation.
    
    Args:
        num (int): The decimal number to convert.
    
    Returns:
        str: The hexadecimal representation of the number.
    """
    hexa = []  # List to store hexadecimal digits

    # Loop until the number becomes 0
    while num != 0:
        # Get the remainder when dividing by 16
        remainder = num % 16

        # Convert remainder to a hexadecimal character
        if remainder < 10:
            # For digits 0-9
            hexa.append(chr(remainder + 48))  # ASCII '0' starts at 48
        else:
            # For letters A-F
            hexa.append(chr(remainder + 55))  # ASCII 'A' starts at 65

        # Update the number by dividing by 16
        num //= 16

    # Reverse the list to get the correct hexadecimal order
    hexa.reverse()

    # Join the list into a string and return
    return ''.join(hexa)


# Prompt the user for input
decimal_num = int(input("Enter a decimal number: "))

# Convert the decimal number to hexadecimal and display the result
hexadecimal = convert_to_hexadecimal(decimal_num)
print(f"Hexadecimal representation of {decimal_num} is: {hexadecimal}")

File: Top 100 Codes/02-Working with Numbers/09-Decimal to Hexadecimal Conversion/DecimaltoHexdecimalConversion.cpp
#include <iostream>
#include <string>
using namespace std;

// Function to convert a decimal number to hexadecimal
string convertToHexadecimal(int decimal)
{
    string hexadecimal = ""; // String to store the hexadecimal result
    int remainder;

    // Characters for hexadecimal digits
    const char hexDigits[] = "0123456789ABCDEF";

    // Loop until the decimal number becomes 0
    while (decimal > 0)
    {
        // Get the remainder when dividing by 16
        remainder = decimal % 16;

        // Append the corresponding hexadecimal digit
        hexadecimal = hexDigits[remainder] + hexadecimal;

        // Divide the decimal number by 16 to process the next digit
        decimal /= 16;
    }

    // Return the hexadecimal representation
    return hexadecimal;
}

int main()
{
    int decimal_num;

    // Prompt the user for input
    cout << "Enter a decimal number: ";
    cin >> decimal_num;

     // Function call to convert the decimal number to hexadecimal
    string hexadecimal = convertToHexadecimal(decimal_num);

    // Display the result
    cout << "The hexadecimal representation of " << decimal_num << " is: " << (hexadecimal.empty() ? "0" : hexadecimal) << endl;

    return 0;
}
